newIDE/app/src/Utils/GDevelopServices/Project.js

```diff
@@ -693,3 +721,80 @@ export const listProjectUserAcls = async (
 
   return projectUserAcls;
 };
+
+export const updateCloudProjectVersion = async (
+  authenticatedUser: AuthenticatedUser,
+  cloudProjectId: string,
+  versionId: string,
+  attributes: {| label: string |}
+): Promise<?CloudProjectVersion> => {
+  const { getAuthorizationHeader, firebaseUser } = authenticatedUser;
+  if (!firebaseUser) return;
+
+  const trimmedLabel = attributes.label.trim();
+
+  const cleanedAttributes = {
+    label: trimmedLabel
+      ? trimmedLabel.slice(0, CLOUD_PROJECT_VERSION_LABEL_MAX_LENGTH)
+      : '',
+  };
+
+  const { uid: userId } = firebaseUser;
+  const authorizationHeader = await getAuthorizationHeader();
+  const response = await apiClient.patch(
+    `/project/${cloudProjectId}/version/${versionId}`,
+    cleanedAttributes,
+    {
+      headers: {
+        Authorization: authorizationHeader,
+      },
+      params: { userId },
+    }
+  );
+  return response.data;
+};
+
+/**
+ * List versions of a cloud project.
+ * This method does not directly use the authenticatedUser object to enable
+ * listing versions in React effects. Using authenticatedUser as a dependency
+ * of an effect triggers the effect on each change of the profile (any update
+ * of badges, extensions, purchases, etc.).
+ */
+export const listVersionsOfProject = async (
+  getAuthorizationHeader: () => Promise<string>,
+  firebaseUser: ?FirebaseUser,
```

Comment:
Nitpicking: if you look at other usages of `getAuthorizationHeader: () =>` in the codebase (Game.js, Build.js, Generation.js, User.js, Shop.js), we usually pass `userId: string` and we prefer to pass `profile.id` for it (which is the same as firebaseUser.uid, but in the future we might use something else than firebase, so better to use our own UserProfile entity).

It's true that `Project.js` and `Play.js` prefer to use authenticatedUser and sadly read firebasUser.uid - but as it's already inconsistent, let's try to prefer profile.id (and avoid passing `firebaseUser`). No need to refactor everything, just this method.

Reply:
Oh yeah you're right